src/lib.rs

```diff
@@ -92,7 +92,7 @@ pub fn compile(cmds: Vec<Cmd>) -> Result<Program, CompileError> {
             // then pops x more values off the stack. Finally, it
             // pushes one value back onto the stack based on the index
             // Thus the net effect of Arr is to reduce the stack size by x.
-            Arr(x) => -(x as isize),
+            Arr(x) => -saturating_as_isize(x),
```

Comment:
This is fine since we'll only lose information about arrays with more that 2^63 elements on them, which isn't practical…